Modified: trunk/CMakeLists.txt
===================================================================
--- trunk/CMakeLists.txt	2012-03-19 20:20:24 UTC (rev 4872)
+++ trunk/CMakeLists.txt	2012-03-27 10:24:53 UTC (rev 4873)
@@ -23,10 +23,22 @@
 # The RC version must always be four comma-separated numbers
 set(RCVERSION 1,2,80,0)

+# Installation paths
+set(BIN_DIR "${CMAKE_INSTALL_PREFIX}/bin")
+set(DATA_DIR "${CMAKE_INSTALL_PREFIX}/share")
+set(MAN_DIR "${DATA_DIR}/man")
+set(LOCALE_DIR "${DATA_DIR}/locale")
+set(DOC_DIR "${CMAKE_INSTALL_PREFIX}/share/doc/${CMAKE_PROJECT_NAME}-${VERSION}")
+
+if(WIN32)
+set(BIN_DIR "${CMAKE_INSTALL_PREFIX}")
+set(DOC_DIR "${CMAKE_INSTALL_PREFIX}")
+endif()
+
 # Compatibility variables for the migration from autotools
 add_definitions(-DPACKAGE_NAME="${CMAKE_PROJECT_NAME}")
 add_definitions(-DPACKAGE_VERSION="${VERSION}")
-add_definitions(-DLOCALEDIR="${CMAKE_INSTALL_PREFIX}/share/locale")
+add_definitions(-DLOCALEDIR="${LOCALE_DIR}")

 if(MSVC)
   message(FATAL_ERROR "TigerVNC cannot be built with Visual Studio.  Please use MinGW")
@@ -162,7 +174,6 @@
 if(ENABLE_NLS)
   # Tools
   find_package(Gettext)
-  set(LOCALE_DIR "${CMAKE_INSTALL_PREFIX}/share/locale")

   # Gettext needs iconv
   find_package(Iconv)

Modified: trunk/cmake/BuildPackages.cmake
===================================================================
--- trunk/cmake/BuildPackages.cmake	2012-03-19 20:20:24 UTC (rev 4872)
+++ trunk/cmake/BuildPackages.cmake	2012-03-27 10:24:53 UTC (rev 4873)
@@ -86,5 +86,5 @@
 # Common
 #

-install(FILES ${CMAKE_SOURCE_DIR}/LICENCE.TXT DESTINATION doc)
-install(FILES ${CMAKE_SOURCE_DIR}/README.txt DESTINATION doc)
+install(FILES ${CMAKE_SOURCE_DIR}/LICENCE.TXT DESTINATION ${DOC_DIR})
+install(FILES ${CMAKE_SOURCE_DIR}/README.txt DESTINATION ${DOC_DIR})

Modified: trunk/java/CMakeLists.txt
===================================================================
--- trunk/java/CMakeLists.txt	2012-03-19 20:20:24 UTC (rev 4872)
+++ trunk/java/CMakeLists.txt	2012-03-27 10:24:53 UTC (rev 4873)
@@ -99,5 +99,5 @@
     ${JAVA_SRCDIR}/README
     ${JAVA_SRCDIR}/index.vnc
     ${JAVA_SRCDIR}/favicon.ico
-    DESTINATION vnc/classes)
+    DESTINATION ${DATA_DIR}/vnc/classes)
 endif()

Modified: trunk/unix/CMakeLists.txt
===================================================================
--- trunk/unix/CMakeLists.txt	2012-03-19 20:20:24 UTC (rev 4872)
+++ trunk/unix/CMakeLists.txt	2012-03-27 10:24:53 UTC (rev 4873)
@@ -4,5 +4,5 @@
 add_subdirectory(vncpasswd)
 add_subdirectory(x0vncserver)

-install(PROGRAMS vncserver DESTINATION bin)
-install(FILES vncserver.man DESTINATION man/man1 RENAME vncserver.1)
+install(PROGRAMS vncserver DESTINATION ${BIN_DIR})
+install(FILES vncserver.man DESTINATION ${MAN_DIR}/man1 RENAME vncserver.1)

Modified: trunk/unix/vncconfig/CMakeLists.txt
===================================================================
--- trunk/unix/vncconfig/CMakeLists.txt	2012-03-19 20:20:24 UTC (rev 4872)
+++ trunk/unix/vncconfig/CMakeLists.txt	2012-03-27 10:24:53 UTC (rev 4873)
@@ -11,5 +11,5 @@

 target_link_libraries(vncconfig tx rfb network rdr ${X11_LIBRARIES})

-install(TARGETS vncconfig DESTINATION bin)
-install(FILES vncconfig.man DESTINATION man/man1 RENAME vncconfig.1)
+install(TARGETS vncconfig DESTINATION ${BIN_DIR})
+install(FILES vncconfig.man DESTINATION ${MAN_DIR}/man1 RENAME vncconfig.1)

Modified: trunk/unix/vncpasswd/CMakeLists.txt
===================================================================
--- trunk/unix/vncpasswd/CMakeLists.txt	2012-03-19 20:20:24 UTC (rev 4872)
+++ trunk/unix/vncpasswd/CMakeLists.txt	2012-03-27 10:24:53 UTC (rev 4873)
@@ -5,5 +5,5 @@

 target_link_libraries(vncpasswd tx rfb rdr os)

-install(TARGETS vncpasswd DESTINATION bin)
-install(FILES vncpasswd.man DESTINATION man/man1 RENAME vncpasswd.1)
+install(TARGETS vncpasswd DESTINATION ${BIN_DIR})
+install(FILES vncpasswd.man DESTINATION ${MAN_DIR}/man1 RENAME vncpasswd.1)

Modified: trunk/unix/x0vncserver/CMakeLists.txt
===================================================================
--- trunk/unix/x0vncserver/CMakeLists.txt	2012-03-19 20:20:24 UTC (rev 4872)
+++ trunk/unix/x0vncserver/CMakeLists.txt	2012-03-27 10:24:53 UTC (rev 4873)
@@ -23,5 +23,5 @@
   message(WARNING "No XTest extension.  x0vncserver will be view-only.")
 endif()

-install(TARGETS x0vncserver DESTINATION bin)
-install(FILES x0vncserver.man DESTINATION man/man1 RENAME x0vncserver.1)
+install(TARGETS x0vncserver DESTINATION ${BIN_DIR})
+install(FILES x0vncserver.man DESTINATION ${MAN_DIR}/man1 RENAME x0vncserver.1)

Modified: trunk/win/vncconfig/CMakeLists.txt
===================================================================
--- trunk/win/vncconfig/CMakeLists.txt	2012-03-19 20:20:24 UTC (rev 4872)
+++ trunk/win/vncconfig/CMakeLists.txt	2012-03-27 10:24:53 UTC (rev 4873)
@@ -9,5 +9,5 @@
 target_link_libraries(vncconfig rfb_win32 rfb Xregion network rdr ws2_32.lib)

 install(TARGETS vncconfig
-  RUNTIME DESTINATION .
+  RUNTIME DESTINATION ${BIN_DIR}
 )

Modified: trunk/win/winvnc/CMakeLists.txt
===================================================================
--- trunk/win/winvnc/CMakeLists.txt	2012-03-19 20:20:24 UTC (rev 4872)
+++ trunk/win/winvnc/CMakeLists.txt	2012-03-27 10:24:53 UTC (rev 4873)
@@ -33,9 +33,9 @@
 endif()

 install(TARGETS winvnc4
-  RUNTIME DESTINATION .
+  RUNTIME DESTINATION ${BIN_DIR}
 )

 install(TARGETS wm_hooks
-  RUNTIME DESTINATION .
+  RUNTIME DESTINATION ${BIN_DIR}
 )
Modified: trunk/vncviewer/CMakeLists.txt
===================================================================
--- trunk/vncviewer/CMakeLists.txt	2012-03-27 10:24:53 UTC (rev 4873)
+++ trunk/vncviewer/CMakeLists.txt	2012-03-27 10:28:18 UTC (rev 4874)
@@ -49,11 +49,7 @@
   target_link_libraries(vncviewer ws2_32)
 endif()

-if(WIN32)
-  install(TARGETS vncviewer DESTINATION .)
-else()
-  install(TARGETS vncviewer DESTINATION bin)
-endif()
+install(TARGETS vncviewer DESTINATION ${BIN_DIR})
 if(UNIX)
-  install(FILES vncviewer.man DESTINATION man/man1 RENAME vncviewer.1)
+  install(FILES vncviewer.man DESTINATION ${MAN_DIR}/man1 RENAME vncviewer.1)
 endif()
