diff -ur tigervnc-1.2.0.orig/CMakeLists.txt tigervnc-1.2.0/CMakeLists.txt
--- tigervnc-1.2.0.orig/CMakeLists.txt	2012-03-09 21:34:29.000000000 +0100
+++ tigervnc-1.2.0/CMakeLists.txt	2012-04-11 14:17:35.621900365 +0200
@@ -284,10 +284,14 @@
 
 option(USE_INCLUDED_FLTK
   "Force the use of the FLTK library bundled with the TigerVNC source")
-if(NOT FLTK_FOUND OR NOT HAVE_FLTK_DEAD_KEYS OR NOT HAVE_FLTK_CLIPBOARD
-  OR NOT HAVE_FLTK_MEDIAKEYS OR NOT HAVE_FLTK_FULLSCREEN
-  OR NOT HAVE_FLTK_CURSOR)
-  set(USE_INCLUDED_FLTK 1)
+if(NOT USE_INCLUDED_FLTK)
+  if(NOT FLTK_FOUND)
+    message(FATAL_ERROR "FLTK not found and not using included FLTK (needed by vncviewer.)")
+  elseif(NOT HAVE_FLTK_DEAD_KEYS OR NOT HAVE_FLTK_CLIPBOARD
+         OR NOT HAVE_FLTK_MEDIAKEYS OR NOT HAVE_FLTK_FULLSCREEN
+         OR NOT HAVE_FLTK_CURSOR)
+    message(WARNING "Some functions not found in system FLTK. vncviewer will not have full functionality.")
+  endif()
 endif()
 if(USE_INCLUDED_FLTK)
   # Check that we have the proper dependencies
