From 2a2f7b76d021a9e8c3ab4aee2360032eb6be6c1a Mon Sep 17 00:00:00 2001
From: Dennis Schridde <devurandom@users.noreply.github.com>
Date: Thu, 29 Nov 2018 10:52:41 +0100
Subject: [PATCH] scripts/haiUtil.cmake: Prepend DESTDIR when creating symlinks

Fixes: #357
---
 scripts/haiUtil.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/haiUtil.cmake b/scripts/haiUtil.cmake
index 417128d..958911e 100644
--- a/scripts/haiUtil.cmake
+++ b/scripts/haiUtil.cmake
@@ -49,7 +49,7 @@ FUNCTION(join_arguments outvar)
 ENDFUNCTION()
 
 macro(srt_install_symlink filepath sympath)
-    install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ${filepath} ${sympath})")
+    install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ${filepath} \\\${DESTDIR}${sympath})")
     install(CODE "message(\"-- Created symlink: ${sympath} -> ${filepath}\")")
 endmacro(srt_install_symlink)
 
