diff --git a/configure b/configure
index d776447..306bcd9 100755
--- a/configure
+++ b/configure
@@ -18846,7 +18846,7 @@ if test "x$use_kde4" = "xyes"; then
   kde4libs=`kde4-config --install lib --expandvars 2>/dev/null`
   kde4incs=`kde4-config --install include --expandvars 2>/dev/null`
 
-  kde4_test_includes="kdecoration.h kcommondecoration.h kdecorationbridge.h plasma/framesvg.h"
+  kde4_test_includes="kdecoration.h kcommondecoration.h plasma/framesvg.h"
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for KWin headers" >&5
 $as_echo_n "checking for KWin headers... " >&6; }
diff --git a/configure.ac b/configure.ac
index dc29bd5..c9ea7b7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -478,7 +478,7 @@ if test "x$use_kde4" = "xyes"; then
   kde4libs=`kde4-config --install lib --expandvars 2>/dev/null`
   kde4incs=`kde4-config --install include --expandvars 2>/dev/null`
 
-  kde4_test_includes="kdecoration.h kcommondecoration.h kdecorationbridge.h plasma/framesvg.h"
+  kde4_test_includes="kdecoration.h kcommondecoration.h plasma/framesvg.h"
 
   dnl Check for KWin headers
   AC_MSG_CHECKING([for KWin headers])
diff --git a/kde/window-decorator-kde4/window.h b/kde/window-decorator-kde4/window.h
index bc26a74..72dde9a 100644
--- a/kde/window-decorator-kde4/window.h
+++ b/kde/window-decorator-kde4/window.h
@@ -24,7 +24,6 @@
 #ifndef _WINDOW_H
 #define _WINDOW_H
 
-#include <kdecorationbridge.h>
 #include <KDE/KActionCollection>
 #include <kdeversion.h>
 
