--- a/gfx/thebes/gfxFont.cpp	2017-06-12 18:37:10.000000000 +0200
+++ b/gfx/thebes/gfxFont.cpp.new	2017-08-01 20:41:55.834344606 +0200
@@ -2637,6 +2637,17 @@
     return sw;
 }
 
+template gfxShapedWord* gfxFont::GetShapedWord(DrawTarget *aDrawTarget, 
+                                               const unsigned char    *aText, 
+                                               uint32_t    aLength, 
+                                               uint32_t    aHash, 
+                                               Script      aRunScript, 
+                                               bool        aVertical, 
+                                               int32_t     aAppUnitsPerDevUnit, 
+                                               gfx::ShapedTextFlags aFlags,
+                                               RoundingFlags aRounding, 
+                                               gfxTextPerfMetrics *aTextPerf GFX_MAYBE_UNUSED);
+
 bool
 gfxFont::CacheHashEntry::KeyEquals(const KeyTypePointer aKey) const
 {
