diff --git a/dockbarx_0.90.3/dockbarx/iconfactory.py b/dockbarx_0.90.3/dockbarx/iconfactory.py
index 3c4323a..2d17b65 100644
--- a/dockbarx_0.90.3/dockbarx/iconfactory.py
+++ b/dockbarx_0.90.3/dockbarx/iconfactory.py
@@ -22,7 +22,10 @@ pygtk.require("2.0")
 import gtk
 import gc
 gc.enable()
-import Image
+try:
+    from PIL import Image
+except ImportError:
+    import Image
 import array
 import cairo
 import gio
diff --git a/dockbarx_0.90.3/dockbarx/theme.py b/dockbarx_0.90.3/dockbarx/theme.py
index 70a108f..3f2c87e 100644
--- a/dockbarx_0.90.3/dockbarx/theme.py
+++ b/dockbarx_0.90.3/dockbarx/theme.py
@@ -24,7 +24,10 @@ import gtk
 import gobject
 import cairo
 import os
-import Image
+try:
+    from PIL import Image
+except ImportError:
+    import Image
 import array
 from common import ODict
 from common import Globals
