diff --git a/vmci-only/linux/driver.c b/vmci-only/linux/driver.c
index 64a3cca..484ba3a 100644
--- a/vmci-only/linux/driver.c
+++ b/vmci-only/linux/driver.c
@@ -1467,8 +1467,13 @@ VMCIUserVALockPage(VA addr) // IN:
    int retval;
 
    down_read(&current->mm->mmap_sem);
-   retval = get_user_pages(current, current->mm, addr,
-                           1, 1, 0, &page, NULL);
+   #if LINUX_VERSION_CODE >= KERNEL_VERSION(4,6,0)
+      retval = get_user_pages(addr,
+         1, 1, 0, &page, NULL);
+   #else
+      retval = get_user_pages(current, current->mm, addr,
+         1, 1, 0, &page, NULL);
+   #endif
    up_read(&current->mm->mmap_sem);
 
    if (retval != 1) {
