diff --git a/vmnet-only/userif.c b/vmnet-only/userif.c
index 4e457ba..7178a13 100644
--- a/vmnet-only/userif.c
+++ b/vmnet-only/userif.c
@@ -113,8 +113,13 @@ UserifLockPage(VA addr) // IN
    int retval;
 
    down_read(&current->mm->mmap_sem);
-   retval = get_user_pages(current, current->mm, addr,
-			   1, 1, 0, &page, NULL);
+   #if LINUX_VERSION_CODE >= KERNEL_VERSION(4,6,0)
+      retval = get_user_pages(addr,
+         1, 1, 0, &page, NULL);
+   #else
+      retval = get_user_pages(current, current->mm, addr,
+         1, 1, 0, &page, NULL);
+   #endif
    up_read(&current->mm->mmap_sem);
 
    if (retval != 1) {
