diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index 9d9dfc8..d61a06c 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -27,11 +27,3 @@ endif()
 #install FTW image
 install (FILES "images/indicator.jpg" DESTINATION ${DATADIR}/indicator-kdeconnect/)
 install (FILES "images/startup.jpg" DESTINATION ${DATADIR}/indicator-kdeconnect/)
-
-#update icons cache
-install(
-	CODE
-	"execute_process (COMMAND gtk-update-icon-cache -t -f ${CMAKE_INSTALL_PREFIX}/share/icons)"
-	CODE
-	"message (STATUS \"Updated icon cache in ${CMAKE_INSTALL_PREFIX}/share/icons\")"
-	)
