diff --git a/stickynotes/Makefile.am b/stickynotes/Makefile.am
index 76eee21..c6143cc 100644
--- a/stickynotes/Makefile.am
+++ b/stickynotes/Makefile.am
@@ -14,6 +14,7 @@ INCLUDES =					\
 	$(GNOME_APPLETS_CFLAGS)			\
 	$(GNOME_LIBS2_CFLAGS)			\
 	$(LIBWNCK_CFLAGS)			\
+	$(X_LIBS)				\
 	$(XML2_CFLAGS)				\
 	-DSTICKYNOTES_ICONDIR=\"$(datadir)/pixmaps/stickynotes/\"
 
@@ -35,6 +36,7 @@ stickynotes_applet_LDADD =	\
 	$(GNOME_APPLETS_LIBS)	\
 	$(LIBWNCK_LIBS)		\
 	$(GNOME_LIBS2_LIBS)	\
+	$(X_LIBS)		\
 	$(XML2_LIBS)
 
 builder_DATA = $(builder_files)
