diff --git a/configure.ac b/configure.ac
index d1502ba..b89ca6b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1950,6 +1950,7 @@ then
   fi
 
   dnl Check for some compiler flags
+  WINE_TRY_CFLAGS([-fno-tree-loop-distribute-patterns])
   WINE_TRY_CFLAGS([-fno-strict-aliasing])
   WINE_TRY_CFLAGS([-fexcess-precision=standard],[AC_SUBST(EXCESS_PRECISION_CFLAGS,"-fexcess-precision=standard")])
   dnl clang needs to be told to fail on unknown options
